src/TorchSharp/Dataset.cs

```diff
@@ -47,11 +45,35 @@ public void Dispose()
                     /// <returns>Tensors of index. DataLoader will catenate these tensors into batches.</returns>
                     public abstract T GetTensor(long index);
 
+                    /// <inheritdoc />
+                    public T this[long index] => GetTensor(index);
+
                     protected virtual void Dispose(bool disposing)
                     {
                     }
                 }
             }
         }
     }
+
+    namespace Modules
+    {
+        /// <summary>
+        /// The base interface for all Datasets.
+        /// </summary>
+        public interface IDataset<out T> : IDisposable
+        {
+            /// <summary>
+            /// Size of dataset
+            /// </summary>
+            long Count { get; }
+
+            /// <summary>
+            /// Get tensor according to index
+            /// </summary>
+            /// <param name="index">Index for tensor</param>
+            /// <returns>Tensors of index. DataLoader will catenate these tensors into batches.</returns>
+            T this[long index] { get; }
```

Comment:
Here we are using `Int64`. However actually most .net containers does not support `Int64` indices. And it actually is not practical to have so much data. Should we switch to `Int32` instead?